pkg/k8sutil/create.go

```diff
@@ -165,3 +172,59 @@ func parseJSONManifest(data []byte) ([]manifest, error) {
 	}
 	return manifests, nil
 }
+
+// CreateOrUpdateNamespace creates the release namespace or updates the namespace
+// if it already exists.
+func CreateOrUpdateNamespace(ns Namespace, nsclient corev1typed.NamespaceInterface) error {
+	if ns.Name == "" {
+		return fmt.Errorf("namespace name can't be empty")
+	}
+
+	_, err := nsclient.Get(context.TODO(), ns.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return createNamespace(ns, nsclient)
+		}
+
+		return fmt.Errorf("getting namespace %q: %w", ns.Name, err)
+	}
+	// Namespace exists, hence updating the namespace.
+	return updateNamespace(ns, nsclient)
+}
+
+func updateNamespace(ns Namespace, nsclient corev1typed.NamespaceInterface) error {
+	// Retrieve the latest version of Namespace before attempting update
+	result, err := nsclient.Get(context.TODO(), ns.Name, metav1.GetOptions{})
```

Comment:
Instead of retrieving here, pass the namespace you got in `CreateOrUpdateNamespace` here as an argument. This way, function is more robust, easier to test and does less.